phpbb/phpbb
# [ticket/16219] 5. CP Common code and tests

Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass _(should eventually pass)_
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket: https://tracker.phpbb.com/browse/PHPBB3-16219

---

This PR is part of a chain of PR's that will move all control panel modules to controllers, that all will be based on each other.
1. [ACP Modules to controllers](https://github.com/phpbb/phpbb/pull/5738)
2. [MCP Modules to controllers](https://github.com/phpbb/phpbb/pull/5739)
3. [UCP Modules to controllers](https://github.com/phpbb/phpbb/pull/5743)
4. [CP Services and items](https://github.com/phpbb/phpbb/pull/5744)
5. [CP Common code and tests](https://github.com/phpbb/phpbb/pull/5746)

---

Todo in other PR's:
- Add tests for new 'control panel system'
- Clean up language strings that are now duplicates 
- Check everything in regards to modules
  - Module table
  - Module functions
  - Module migrations ???
